x-pack/legacy/plugins/apm/server/lib/apm_telemetry/types.ts

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { DeepPartial } from 'utility-types';
+import { AgentName } from '../../../common/agent_name';
+
+export interface TimeframeMap {
+  '1d': number;
+  '1M': number;
+  '6M': number;
+  all: number;
+}
+
+export type TimeframeAll = Pick<TimeframeMap, 'all'>;
+
+export type APMDataTelemetry = DeepPartial<{
+  has_any_services: boolean;
+  services_per_agent: Record<AgentName, number>;
+  versions: {
+    apm_server: {
+      minor: number;
+      major: number;
+      patch: number;
+    };
+  };
+  counts: {
+    transaction: TimeframeMap;
+    span: TimeframeMap;
+    error: TimeframeMap;
+    metric: TimeframeMap;
+    sourcemap: TimeframeMap;
+    onboarding: TimeframeMap;
+    agent_configuration: TimeframeAll;
+    max_transaction_groups_per_service: TimeframeAll;
+    max_error_groups_per_service: TimeframeAll;
+    traces: TimeframeAll;
+    services: TimeframeAll;
+  };
+  integrations: {
+    ml: {
+      has_anomalies_indices: boolean;
+    };
+  };
+  agents: Record<
+    AgentName,
+    {
+      agent: {
+        version: string[];
+      };
+      service: {
+        framework: {
+          name: string[];
+          version: string[];
+        };
+        language: {
+          name: string[];
+          version: string[];
+        };
+        runtime: {
+          name: string[];
+          version: string[];
+        };
+      };
+    }
+  >;
+  indices: {
+    shards: {
+      total: number;
+    };
+    all: {
+      total: {
+        docs: {
+          count: number;
+        };
+        store: {
+          size_in_bytes: number;
+        };
+      };
+    };
+  };
+  tasks: Record<
+    | 'processor_events'
+    | 'agent_configuration'
+    | 'services'
+    | 'versions'
+    | 'groupings'
+    | 'integrations'
+    | 'agents'
+    | 'indices_stats',
+    { took: { ms: number } }
+  >;
+}>;
```

Comment:
It's amazing to have this! This will help us much to create the mapping in the telemetry cluster! Thanks for being so thoughtful and strict with the types! ❤️